proposals/p1964.md

```diff
@@ -0,0 +1,134 @@
+# Character literals
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/1964)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+    -   [Types](#types)
+    -   [Encoding](#encoding)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+
+<!-- tocstop -->
+
+## Problem
+
+This proposal specifies lexical rules for constant characters in Carbon.
+
+## Background
+
+We wish to provide a distinct lexical syntax for character literals versus
+string literals.
+
+In theory we could just reuse string literals for the purpose of character
+literals. However, it could benefit the readablity of our code if we had a
+distinct lexical syntax for character literals versus string literals.
+
+## Proposal
+
+The idea is to create and manage a character literal the same we would as a
+string, but using the single quote delimiter (') compared to the string double
+quote (").
+
+As with string literals, each character literal would have a different type.
+
+    var w: ch8 = 'w';
+
+We will not support:
+
+-   Multi-line literals
+-   "raw" literals (using #'x'#)
+-   Empty character literals (''')
+
+## Details
```

Comment:
This seems a bit different from what was proposed in #1934 . Is that intentional?  In #1934 :
- the type includes the _value_ of the character so `'b'` and `'c'` are different types
- there are no prefixes like `u`, `U`, or `L`
- the difference between 8-bit characters and 16-bit characters is from the type of the variable you assign it to (and the legal literals that can be assigned to that variable), not anything in the literal syntax.

Reply:
Oh I see,  will fix this